WordPress/gutenberg
# [RNMobile] Remove title focus for default editor view

## Related PRs

* `Gutenberg Mobile:` https://github.com/wordpress-mobile/gutenberg-mobile/pull/4740
* `iOS:` https://github.com/wordpress-mobile/WordPress-iOS/pull/18338

## What?

By default, when a user first opens the post editor within the WordPress mobile app, the title is focused. This PR changes that and makes it so that there is no default focus when opening the editor.

## Why?

User research indicates that it's not always clear to users what primary actions are available when they first open the editor. By iterating on the default view of the editor, it's hoped that it will be easier for users to discover blocks and, therefore, improve the overall writing/onboarding experience.

This PR is part of a series of changes we're making to the editor's default view within the app. Another, related to change to the way the block inserter button is styled within the default view can be found in https://github.com/WordPress/gutenberg/pull/39726.

## How?

All references to `setFocusOnTitle` have been removed from the Gutenberg and iOS codebase. Note, it wasn't necessary to remove any references from the Android codebase, as all Android-related references exist within Gutenberg. 

## Testing Instructions

* When opening the post editor within the iOS or Android apps, verify that the title is _not_ focused by default.
* Experiment with various writing and publishing flows as part of testing. For example, make sure that the title _isn't_ focused when saving and returning to a post, verify that there are no regressions when going through [the writing flow test cases](https://github.com/wordpress-mobile/test-cases/edit/trunk/test-cases/gutenberg/writing-flow/readme.md), etc. The title field should only be focused when adding text.

### Possible Regressions 

Possible regressions involve the title not being focused during times it should, so it'd be especially helpful to verify that the title remains in focus during key writing flows, such as switching between portrait/landscape while typing a title.

## Screenshots or screencast 

| Before (Title Focus)  | After (No Focus) |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/162721314-4d21a8a4-2fb6-47ff-bb08-7c01fcb721cd.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/162721244-4d461f96-d837-484b-b8fd-a367453d8707.png" width="100%"> |
